spark_log_parser/__main__.py

```diff
@@ -1,37 +1,42 @@
+import argparse
 import logging
+import os
+import sys
+import tempfile
+from pathlib import Path
 
 logging.captureWarnings(True)
 
-from spark_log_parser.parsing_models.application_model_v2 import sparkApplication
-
-import os
-import argparse
-import sys
+from spark_log_parser.eventlog import EventLog  # noqa: E402
+from spark_log_parser.parsing_models.application_model_v2 import sparkApplication  # noqa: E402
 
 logger = logging.getLogger("spark_log_parser")
-            
-parser = argparse.ArgumentParser("spark_log_parser")
-parser.add_argument("-l", "--log-file", required=True, help="path to event log")
-parser.add_argument("-r", "--result-dir", required=True, help="path to directory in which to save parsed logs")
-args = parser.parse_args()  
-
-print("\n" + "*" * 12 + " Running the Log Parser for Spark Predictor " + "*" * 12 + "\n")
 
-log_path = os.path.abspath(args.log_file)
+parser = argparse.ArgumentParser("spark_log_parser")
+parser.add_argument("-l", "--log-file", required=True, type=Path, help="path to event log")
+parser.add_argument(
+    "-r", "--result-dir", required=True, help="path to directory in which to save parsed logs"
+)
+args = parser.parse_args()
 
 if not os.path.isdir(args.result_dir):
     logger.error("%s is not a directory", args.result_dir)
     sys.exit(1)
 
-print("\n--Processing log file: " + log_path)
+print("\n" + "*" * 12 + " Running the Log Parser for Spark Predictor " + "*" * 12 + "\n")
+print("--Processing log file: " + str(args.log_file))
 
-log_name = os.path.basename(log_path)
-result_path = os.path.join(args.result_dir, "parsed-" + log_name)
+with tempfile.TemporaryDirectory() as work_dir:
+    event_log = EventLog(source_url=args.log_file.resolve().as_uri(), work_dir=work_dir)
+    app = sparkApplication(eventlog=str(event_log.event_log))
 
-if os.path.exists(result_path):
-    os.remove(result_path)
+if args.log_file.suffixes:
+    result_path = os.path.join(
+        args.result_dir, "parsed-" + args.log_file.name[: -len("".join(args.log_file.suffixes))]
```

Comment:
Clever way of stripping suffixes!

Reply:
Yeah, kinda wish there was something in pathlib for this